buffer/src/main/java/io/netty/buffer/PoolSubpage.java

```diff
@@ -115,7 +115,9 @@ boolean free(PoolSubpage<T> head, int bitmapIdx) {
 
         if (numAvail ++ == 0) {
             addToPool(head);
-            return true;
+            if (maxNumElems > 1) {
```

Comment:
@ZzxyNn can you just please add a comment here any explain why we return true only if `maxNumElems > 1` ? This will make it easier to understand this in the future.